PYME/experimental/labview_spooling_hacks.py

```diff
@@ -30,7 +30,8 @@ def spoof_focus_from_metadata(mdh):
             'StackSettings.NumCycles'], mdh['StackSettings.FramesPerStep'])
         position = np.arange(mdh.getOrDefault('Protocol.PiezoStartPos', 0),
                              mdh.getOrDefault('Protocol.PiezoStartPos', 0) + mdh['StackSettings.NumSteps'] * mdh[
-                                 'StackSettings.StepSize'], mdh['StackSettings.StepSize'])
+                                 'StackSettings.StepSize'], (mdh['StackSettings.StepSize'] 
+                                                             if mdh['StackSettings.StepSize'] else 1))
```

Comment:
What is the rationale here? - it looks like you are just replacing one nonsense value with another nonsense value. If there is really a problem with people giving you bad step sizes it would surely be better to do something like `assert(mdh['StackSettings.StepSize'] > 0)` before this line so that they get an error message that makes sense. As it is you are replacing a value you are making a logical comparison which doesn't make sense (`mdh['StackSettings.StepSize']` is not a bool) and then replacing with a value which is guaranteed to be wrong and hoping for the best.

Reply:
This change just forces this line to throw an error that the `try` / `except` block is designed to handle (the `assert` fails). Prior to this change, this line would throw a `ZeroDivisionError` or a `ValueError` or a `RuntimeWarning` depending on how numpy felt that particular run. `assert(mdh['StackSettings.StepSize'] > 0)` is a better way of handling this. I can cheat and use `mdh['StackSettings.StepSize']` as a bool here thanks to duck typing, since it's zero or a positive number. It's not pretty, but it's technically OK.